odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/AlertRepository.java

```diff
@@ -1,27 +1,108 @@
 package org.opendatadiscovery.oddplatform.repository;
 
 import java.util.Collection;
+import java.util.List;
+import java.util.Set;
 import org.opendatadiscovery.oddplatform.dto.alert.AlertDto;
 import org.opendatadiscovery.oddplatform.dto.alert.AlertStatusEnum;
 import org.opendatadiscovery.oddplatform.model.tables.pojos.AlertPojo;
+import org.opendatadiscovery.oddplatform.repository.reactive.ReactiveCRUDRepository;
 import org.opendatadiscovery.oddplatform.utils.Page;
+import reactor.core.publisher.Mono;
 
-public interface AlertRepository {
-    Page<AlertDto> listAll(final int page, final int size);
+public interface AlertRepository extends ReactiveCRUDRepository<AlertPojo> {
 
-    Page<AlertDto> listByOwner(final int page, final int size, final long ownerId);
+    /**
+     * Retrieves all alerts with status AlertStatusEnum.OPEN.
+     *
+     * @param page - page offset
+     * @param size - amount of retrieved entries
+     * @return - {@link Page} of {@link AlertDto}
+     */
+    Mono<Page<AlertDto>> listAllWithStatusOpen(final int page, final int size);
 
-    Collection<AlertDto> getDataEntityAlerts(final long dataEntityId);
+    /**
+     * Retrieves all alerts with status AlertStatusEnum.OPEN for certain owner.
+     *
+     * @param page    - page offset
+     * @param size    - amount of retrieved entries
+     * @param ownerId - owner id
+     * @return - {@link Page} of {@link AlertDto}
+     */
+    Mono<Page<AlertDto>> listByOwner(final int page, final int size, final long ownerId);
 
-    Page<AlertDto> listDependentObjectsAlerts(final int page, final int size, final long ownerId);
+    /**
+     * Retrieves all alerts with status AlertStatusEnum.OPEN for certain DataEntity.
+     *
+     * @param dataEntityId - data entity id
+     * @return - List of {@link AlertDto}
+     */
+    Mono<List<AlertDto>> getAlertsByDataEntityId(final long dataEntityId);
```

Comment:
@damirabdul, @Chismur, gents, totally unimportant, but wouldn't we want to return Flux in here?

Reply:
as for me doesn't matter, because we need List for mapping, therefore need to do invoke collectList();